### ISSUE TITLE
BUG: query modifies the frame when you compare with `=`

### ISSUE THREAD

**Issue by TomAugspurger** [pydata/pandas#8664]
I messed up and used `=` instead of `==` in a `query`.

```python
df = pd.DataFrame({'a': [1, 2, 3], 'b': ['a', 'b', 'c']})
df.query('a=1')
```

That raises a ValueError. But `df` was modified.

```python
In [15]: df
Out[15]:
   a  b
0  1  a
1  1  b
2  1  c
```

versions:

```
pandas: 0.15.0-6-g403f38d
bottleneck: None
tables: None
numexpr: 2.3.1
```

Can't look right now.

**Comment by kohaugustine** [comment]
Hi @jorisvandenbossche , I'm here now on Github! I last communicated with you about this bug on Stackoverflow http://stackoverflow.com/questions/28714469/bug-in-pandas-query-method/ about the bug wherein the query() method would modify the data in a column if the query statement used an '=' symbol. I'd be keen to contribute to fixing this bug, however, if I do work on it, I can't guarantee to a strict timeline and may not have a lot of time to commit to it especially since at work I have moved on to a slightly different assignment where I am not doing much data analysis and thus not using pandas anymore. I also have a fair amount of other projects outside of work that I have already undertaken. But I would say that I enjoy using python and am quite familar with it, so if solving this bug is within my reach, I definitely like to contribute to this !

**Comment by kohaugustine** [comment]
May I know what are the steps I need to take to be able to start assisting with this debug? Like do I have to clone the entire development repo and then create a branch or something? What's the flow like at the moment?

**Comment by jreback** [comment]
see docs [here](https://github.com/pydata/pandas/wiki/Contributing)

- you fork the repo (only do this once)
- create a branch
- add tests which repro the problem
- make a fix
- tests should now pass
- submit a pull-request

**Comment by kohaugustine** [comment]
@jreback  Thank you for pointing me to the docs and for describing the flow. I'll set to work on this, and if I have any issues, I'll raise them up to you guys.

**Comment by jreback** [comment]
closed by #11149
